src/Views/Textile/Step.elm

```diff
@@ -185,6 +187,59 @@ dyeingMediumField { inputs, updateDyeingMedium } =
         ]
 
 
+spinningProcessField : Config msg -> Html msg
+spinningProcessField { inputs, updateMaterialSpinning } =
+    li [ class "list-group-item d-flex align-items-center gap-2" ]
+        [ div [ class "d-flex flex-column gap-1 w-100" ]
+            (inputs.materials
+                |> List.map
+                    (\{ material, spinning } ->
+                        let
+                            availableSpinningProcesses =
+                                Material.getAvailableSpinningProcesses material.origin
+                        in
+                        div [ class "d-flex justify-content-between align-items-center fs-7" ]
+                            [ label
+                                [ for <| "spinning-for-" ++ Material.idToString material.id
+                                , class "text-truncate w-25"
+                                ]
+                                [ text material.shortName ]
+                            , case availableSpinningProcesses of
+                                [ spinningProcess ] ->
+                                    input
+                                        [ type_ "text"
+                                        , class "form-control w-75"
+                                        , disabled True
+                                        , value <| Material.spinningToLabel spinningProcess
+                                        ]
+                                        []
+
+                                _ ->
+                                    availableSpinningProcesses
+                                        |> List.map
+                                            (\spinningProcess ->
+                                                option
+                                                    [ value <| Material.spinningToString spinningProcess
+                                                    , selected <| Just spinningProcess == spinning
+                                                    ]
+                                                    [ text <| Material.spinningToLabel spinningProcess
+                                                    ]
+                                            )
```

Comment:
I'm surprised we don't use the extracted value here, which could (should?) be the available processes. I'm trying to wrap my head around why it's done that way, though if unavoidable a comment would be welcome.

Reply:
Yeah, I have the extracted value already in the `let..in` block above, but you're right I could move it in the `case..of` and then use the extract values, maybe that's more readable 👍